ns1/resource_record.go

```diff
@@ -261,47 +270,54 @@ func resourceDataToRecord(r *dns.Record, d *schema.ResourceData) error {
 	if v, ok := d.GetOk("link"); ok {
 		r.LinkTo(v.(string))
 	}
-	// if v, ok := d.GetOk("meta"); ok {
-	// 	metaDynamicToStruct(r.Meta, v)
-	// }
+
+	if v, ok := d.GetOk("meta"); ok {
+		log.Println("record meta", v)
+		r.Meta = data.MetaFromMap(v.(map[string]interface{}))
+		log.Println(r.Meta)
+	}
 	useClientSubnet := d.Get("use_client_subnet").(bool)
 	r.UseClientSubnet = &useClientSubnet
 
 	if rawFilters := d.Get("filters").([]interface{}); len(rawFilters) > 0 {
-		f := make([]*filter.Filter, len(rawFilters))
+		filters := make([]*filter.Filter, len(rawFilters))
 		for i, filterRaw := range rawFilters {
 			fi := filterRaw.(map[string]interface{})
 			config := make(map[string]interface{})
-			filter := filter.Filter{
+			f := filter.Filter{
 				Type:   fi["filter"].(string),
 				Config: config,
 			}
 			if disabled, ok := fi["disabled"]; ok {
-				filter.Disabled = disabled.(bool)
+				f.Disabled = disabled.(bool)
 			}
 			if rawConfig, ok := fi["config"]; ok {
 				for k, v := range rawConfig.(map[string]interface{}) {
 					if i, err := strconv.Atoi(v.(string)); err == nil {
-						filter.Config[k] = i
+						f.Config[k] = i
 					} else {
-						filter.Config[k] = v
+						f.Config[k] = v
 					}
 				}
 			}
-			f[i] = &filter
+			filters[i] = &f
 		}
-		r.Filters = f
+		r.Filters = filters
 	}
-	if regions := d.Get("regions").(*schema.Set); regions.Len() > 0 {
-		for _, regionRaw := range regions.List() {
+	if regions := d.Get("regions").([]interface{}); len(regions) > 0 {
+		for _, regionRaw := range regions {
 			region := regionRaw.(map[string]interface{})
 			ns1R := data.Region{
 				Meta: data.Meta{},
 			}
-			// if v, ok := region["meta"]; ok {
-			// 	metaDynamicToStruct(&ns1R.Meta, v)
-			// }
 
+			if v, ok := region["meta"]; ok {
+				log.Println("region meta", v)
```

Comment:
should we leave the log statements?

Reply:
Does it hurt to leave them in there? People usually don't have TF_LOG set when they're running stuff and it's helpful for dev. That being said in the final version we want to merge I don't mind removing them.

Reply:
Oh ya, thats a good point about TF_LOG. in that case i agree, these statements seem useful for debugging